### ISSUE TITLE
gsutil ls -lR returns nothing if there is a trailing "/"

### ISSUE THREAD

**Issue by smrgit** [GoogleCloudPlatform/gsutil#367]
for example (and this is an open-access bucket):
```
bash-4.2$  gsutil ls -lR gs://isb-cgc-open/ccle/
         0  2016-03-11T17:42:23Z  gs://isb-cgc-open/ccle/
TOTAL: 1 objects, 0 bytes (0 B)
```

whereas without the trailing "/" it behaves as expected:
```
bash-4.2$ gsutil ls -lR gs://isb-cgc-open/ccle
gs://isb-cgc-open/ccle/:
         0  2016-03-11T17:42:23Z  gs://isb-cgc-open/ccle/
  12552842  2016-03-11T17:42:23Z  gs://isb-cgc-open/ccle/1650_HC_plus_RD_muts.maf.annotated.gz
    325803  2016-03-11T17:42:25Z  gs://isb-cgc-open/ccle/CCLE_Oncomap3_2012-04-09.maf
  66444623  2016-03-11T17:42:25Z  gs://isb-cgc-open/ccle/CCLE_expression_CN_muts_GENEE_2010-04-16.gctx
<snip>
```

**Comment by thobrla** [comment]
The issue here is that there exists an object named `isb-cgc-open/ccle/`, so you have both an object and a "directory" with that name.  In a filesystem this could never occur; gsutil assumes you were referring to the object.

**Comment by smrgit** [comment]
However, the GCS browser (in the Console) seems to understand that the "object" named ``isb-cgc-open/ccle/`` is of type "Folder", but gsutil does not?

**Comment by thobrla** [comment]
Yes - the GCS browser uses such placeholder objects to provide the illusion of empty folders (which gsutil does not support).

**Comment by smrgit** [comment]
It still seems like it would be logical for ``ls -lR`` to be able to see/list "through" ``gs://isb-cgc-open/ccle/`` but I'm getting the impression that there is not likely to be any movement on this.

**Comment by thobrla** [comment]
I agree that it's not optimal since it would be nice if gsutil interacted seamlessly with the browser's placeholder object.  How much pain is this causing you, though?  Is it hard to omit the trailing slash?  I'd be willing to accept a pull request on this but at first glance it doesn't seem like a high priority fix.  Feel free to help me understand otherwise.
